optional.py

```diff
@@ -16,6 +16,21 @@ def of(thing):
     def empty():
         return Optional(None)
 
+    def __eq__(self, other):
+        if not isinstance(other, Optional):
+            return False
+
+        if self.is_empty() and other.is_empty():
+            return True
+
+        if self.is_empty() or other.is_empty():
+            return False
+
+        return other.get() == self.get()
+
+    def __neq__(self, other):
```

Comment:
I don't think this is right. I think the actual dunder method is ```__ne__``` and I think this is the default implementation. Tests should pass with this method removed.

Reply:
removed.